File: leetcode/easy/14. Longest Common Prefix.ts
function longestCommonPrefix(strs: string[]): string {
    if (strs.length === 1) return strs[0];
    let longest_prefix: string = "";
    const first_word: string = strs[0];
    for (let length = 1; length <= first_word.length; length += 1){
        const temp_prefix: string = first_word.substring(0, length);
        // console.log("temp PREFIX:",temp_prefix);
        const regex: RegExp = new RegExp(`^${temp_prefix}`)
        for (let i = 1; i < strs.length; i += 1){
            if (strs[i].length < length) return longest_prefix;
            if (!regex.test(strs[i])) return longest_prefix;
        }
        longest_prefix = temp_prefix;
    }
    return longest_prefix;
};

File: leetcode/easy/14. Longest Common Prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let mut longest_prefix: String = String::from("");
        let first_word: String = strs[0].clone();
        for length in 1..first_word.len()+1{
            let temp_prefix: &str = &first_word[0..length];
            // println!("prefix: {}", temp_prefix);
            for i in 1..strs.len(){
                if strs[i].len() < length{
                    return longest_prefix;
                }
                if temp_prefix != &strs[i][0..length]{
                    return longest_prefix;
                }
            }
            longest_prefix = String::from(temp_prefix);
        }
        return longest_prefix;
    }
}
